DolbyIO/Source/DolbyIOModule/Private/DolbyIO/SdkAccess.cpp

```diff
@@ -185,64 +122,104 @@ namespace Dolby
 			        }
-			        Observer->OnListOfActiveSpeakersChanged(ActiveSpeakers);
+			        Observer.OnActiveSpeakersChangedEvent(ActiveSpeakers);
 		        })
 		    .on_error(MakeHandler(__LINE__));
+
+		Observer.OnInitializedEvent();
 	}
 	catch (...)
 	{
 		MakeHandler(__LINE__).RethrowAndUpdateStatus();
 	}
 
-	FErrorHandler FSdkAccess::MakeHandler(int Id)
+	bool FSdkAccess::IsConnected() const
 	{
-		return FErrorHandler([this, Id](const FMessage& Msg)
-		                     { Status.SetMsg(Msg + " {" + std::to_string(Id).c_str() + "}"); },
-		                     [this]()
-		                     {
-			                     if (Status.IsConnected())
-			                     {
-				                     Status.SetStatus(EConferenceStatus::leaving);
-				                     Disconnect();
-			                     }
-		                     });
+		return ConferenceStatus == EConferenceStatus::joined;
 	}
 
-	void FSdkAccess::Connect(const FToken& Token, const FConferenceName& Conf, const FUserName& User)
+	namespace
+	{
+		using EConferenceStatus = FSdkAccess::EConferenceStatus;
+
+		FString ToString(EConferenceStatus Status)
+		{
+			switch (Status)
+			{
+				case EConferenceStatus::creating:
+					return "creating";
+				case EConferenceStatus::created:
+					return "created";
+				case EConferenceStatus::joining:
+					return "joining";
+				case EConferenceStatus::joined:
+					return "joined";
+				case EConferenceStatus::leaving:
+					return "leaving";
+				case EConferenceStatus::left:
+					return "left";
+				case EConferenceStatus::destroyed:
+					return "destroyed";
+				case EConferenceStatus::error:
+					return "error";
+				default:
+					return "unknown";
+			};
+		}
+	}
+
+	void FSdkAccess::UpdateStatus(EConferenceStatus Status)
+	{
+		ConferenceStatus = Status;
+		UE_LOG(LogDolbyIO, Log, TEXT("Conference status: %s"), *ToString(ConferenceStatus));
+
+		switch (ConferenceStatus)
+		{
+			case EConferenceStatus::joined:
+				Observer.OnConnectedEvent(LocalParticipantID);
+				break;
+			case EConferenceStatus::left:
+			case EConferenceStatus::error:
+				Observer.OnDisconnectedEvent();
+				break;
+		}
+	}
+
+	void FSdkAccess::Connect(const FString& ConferenceName, const FString& UserName)
 	try
 	{
 		if (!Sdk)
 		{
-			Initialize(Token);
-			if (!Sdk)
-			{
-				return Status.SetMsg("Enter valid token");
-			}
+			UE_LOG(LogDolbyIO, Warning, TEXT("Cannot connect - not initialized"));
+			return;
 		}
-		if (Conf.IsEmpty() || User.IsEmpty())
+		if (IsConnected())
 		{
-			return Status.SetMsg("Conference name and user name cannot be empty");
+			UE_LOG(LogDolbyIO, Warning, TEXT("Cannot connect - already connected, please disconnect first"));
+			return;
 		}
-		if (Status.IsConnected())
+		if (ConferenceName.IsEmpty())
 		{
-			return Status.SetMsg("Must disconnect first");
+			UE_LOG(LogDolbyIO, Warning, TEXT("Cannot connect - conference name cannot be empty"));
+			return;
 		}
 
-		bIsDemo = Conf == "demo";
```

Comment:
This seems like a hack, probably best to expose the demo conference flag as an option explicitly and explain to the developer what it is. Essentially, demo conference is a conference that automatically brings in 3 other bot users into the conference so it's a quick way to validate the connection to the service with audio functionality.

Reply:
This is basically only for internal plugin development purposes, but if such functionality could be considered useful for external users, then I would suggest adding a ConnectToDemoConference function to the plugin's API. We could then easily avoid these hacky shenanigans.

Reply:
Added such a function. Please verify if this is a change we want.